lib/LaTeXML/Core/Gullet.pm

```diff
@@ -338,47 +335,47 @@ sub readXToken {
           $self->handleMarker($token); } } }
     if (!defined $token) {
       return unless $$self{autoclose} && $toplevel && @{ $$self{mouthstack} };
-      $self->closeMouth; }                                       # Next input stream.
-    elsif (my $unexpanded = $$token[2]) {                        # Inline get_dont_expand
-      return $token; }                                           # Defer expansion (recursion?)
-                                                                 # Wow!!!!! See TeX the Program \S 309
+      $self->closeMouth; }    # Next input stream.
+        # Handle \noexpand and  smuggled tokens; either expand to $$token[2] or defer till later
+    elsif (my $unexpanded = $$token[2]) {    # Inline get_dont_expand
+      return ($cc != CC_SMUGGLE_THE) || $LaTeXML::SMUGGLE_THE ? $token : $unexpanded; }
+    ## Wow!!!!! See TeX the Program \S 309
     elsif (!$LaTeXML::ALIGN_STATE    # SHOULD count nesting of { }!!! when SCANNED (not digested)
       && $LaTeXML::READING_ALIGNMENT
       && (($atoken, $atype, $ahidden) = $self->isColumnEnd($token))) {
       $self->handleTemplate($LaTeXML::READING_ALIGNMENT, $token, $atype, $ahidden); }
-    ## Note: special-purpose lookup in State, for efficiency
-    elsif (defined($defn = LaTeXML::Core::State::lookupExpandable($STATE, $token, $toplevel))) {
+    ## Note: use general-purpose lookup, since we may reexamine $defn below
+    elsif ($LaTeXML::Core::State::CATCODE_ACTIVE_OR_CS[$$token[1]]
+      && defined($defn = LaTeXML::Core::State::lookupMeaning($STATE, $token))
```

Comment:
also, unsure if there are performance implications, but it's a bit briefer to write `$STATE->lookupMeaning` than the explicit Perl 4 namespaced subroutine call. But I wouldn't be surprised if you reply that you've measured and the arrow syntax is somehow slower :>